src/plugin.ts

```diff
@@ -32,6 +32,8 @@ export class PluginTypeEnvironment extends TypeEnvironment {
 
     public drawableTypes: Map<CoreElementKind, ObjectType>;
 
+    public startTypes: [Type[], Type][];
```

Comment:
It's addressed in my WIP branch. startTypes will stay the same but it will be wrapped again by computing the args and the return type for you.